Mathlib/AlgebraicTopology/SimplicialSet/NerveAdjunction.lean

```diff
@@ -0,0 +1,671 @@
+      id_comp]
+    exact φ.map_comp (X := (0 : Fin 3)) (Y := 1) (Z := 2)
+      (homOfLE (by decide)) (homOfLE (by decide))
+
+@[simp]
+theorem nerve₂Adj.counit.app_eq (C : Cat) :
+    SSet.Truncated.HomotopyCategory.quotientFunctor (nerveFunctor₂.obj C) ⋙
+      nerve₂Adj.counit.app.{u} C =
+    (whiskerRight OneTruncation₂.ofNerve₂.natIso.hom _ ≫
+      ReflQuiv.adj.{u}.counit).app C := rfl
+
+/-- Naturality of `nerve₂Adj.counit.app` is proven using `HomotopyCategory.lift_unique'`. -/
+theorem nerve₂Adj.counit.naturality ⦃C D : Cat.{u, u}⦄ (F : C ⟶ D) :
+    (nerveFunctor₂ ⋙ hoFunctor₂).map F ⋙ nerve₂Adj.counit.app D =
+      nerve₂Adj.counit.app C ⋙ F := by
+  apply HomotopyCategory.lift_unique'
+  have := hoFunctor₂_naturality (nerveFunctor₂.map F)
+  conv => lhs; rw [← Functor.assoc]; lhs; apply this.symm
+  simp only [Cat.freeRefl_obj_α, ReflQuiv.of_val, comp_obj, Functor.comp_map]
+  rw [← Functor.assoc _ _ F]
+  conv => rhs; lhs; exact (nerve₂Adj.counit.app_eq C)
+  conv => rhs; exact ((whiskerRight OneTruncation₂.ofNerve₂.natIso.hom Cat.freeRefl ≫
+    ReflQuiv.adj.counit).naturality F).symm
+  simp only [app, Cat.comp_eq_comp, Functor.comp_map, Functor.assoc,
+    SSet.Truncated.HomotopyCategory.quotientFunctor]
+  rw [Quotient.lift_spec]
+
+/-- The counit of `nerve₂Adj.` -/
+def nerve₂Adj.counit : nerveFunctor₂ ⋙ hoFunctor₂.{u} ⟶ (𝟭 Cat) where
+  app := nerve₂Adj.counit.app
+  naturality := nerve₂Adj.counit.naturality
+
+local notation (priority := high) "[" n "]" => SimplexCategory.mk n
+
+/-- Because nerves are 2-coskeletal, the components of a map of 2-truncated simplicial sets valued
+in a nerve can be recovered from the underlying ReflPrefunctor. -/
+def toNerve₂.mk.app {X : SSet.Truncated 2} {C : Cat}
+    (F : SSet.oneTruncation₂.obj X ⟶ ReflQuiv.of C)
+    (n : SimplexCategory.Truncated 2) :
+    X.obj (op n) ⟶ (nerveFunctor₂.obj C).obj (op n) := by
+  obtain ⟨n, hn⟩ := n
+  induction' n using SimplexCategory.rec with n
+  match n with
+  | 0 => exact fun x => .mk₀ (F.obj x)
+  | 1 => exact fun f => .mk₁ (F.map ⟨f, rfl, rfl⟩)
+  | 2 => exact fun φ => .mk₂ (F.map (ev01₂ φ)) (F.map (ev12₂ φ))
+
+@[simp] theorem toNerve₂.mk.app_zero {X : SSet.Truncated 2} {C : Cat}
+    (F : SSet.oneTruncation₂.obj X ⟶ ReflQuiv.of C) (x : X _[0]₂) :
+    mk.app F [0]₂ x = .mk₀ (F.obj x) := rfl
+
+@[simp] theorem toNerve₂.mk.app_one {X : SSet.Truncated 2} {C : Cat}
+    (F : SSet.oneTruncation₂.obj X ⟶ ReflQuiv.of C) (f : X _[1]₂) :
+    mk.app F [1]₂ f = .mk₁ (F.map ⟨f, rfl, rfl⟩) := rfl
+
+@[simp] theorem toNerve₂.mk.app_two {X : SSet.Truncated 2} {C : Cat}
+    (F : SSet.oneTruncation₂.obj X ⟶ ReflQuiv.of C) (φ : X _[2]₂) :
+    mk.app F [2]₂ φ = .mk₂ (F.map (ev01₂ φ)) (F.map (ev12₂ φ)) := rfl
+
+/-- This is similiar to one of the famous Segal maps, except valued in a product rather than a
+pullback.-/
+noncomputable def nerve₂.seagull (C : Cat.{v, u}) :
```

Comment:
```suggestion
/-- This is similar to one of the famous Segal maps, except valued in a product rather than a
pullback.-/
noncomputable def nerve₂.seagull (C : Cat.{v, u}) :
```
`seagull` isn't an established name in the literature is it? :open_mouth: :laughing: